src/application/service/GoogleCalendarEventsService.ts

```diff
@@ -39,119 +50,111 @@ export default class GoogleCalendarEventsService {
+                )
                 .map((item) => ({
-                    summary: `[${item.summary}](${item.htmlLink})`,
+                    summary: item.summary ? `[${item.summary}](${item.htmlLink})` : `[No title](${item.htmlLink})`,
                     start: this.parseDateToList(
                         item.start.dateTime,
-                        item.start.timeZone,
                     ),
                     end: this.parseDateToList(
                         item.end.dateTime,
-                        item.end.timeZone,
                     ),
+                    hangoutLink: item.htmlLink,
+                    day: today,
+                    eventDate: item.start.dateTime,
+                    when: `${this.parseDateToList(
+                        item.start.dateTime,
+                    )} - ${this.parseDateToList(item.end.dateTime)}`,
+                    where: `${item.location ?? item.hangoutLink ?? ''}`,
+                    guests: `${this.organizeGuests(
+                        item.creator,
+                        item.attendees,
+                    )}`,
                 })),
         };
     }
 
-    private async getCalendarEventsList() {
-        // const calendarList = await this.api.getCalendarList();
-        const calendarId = this.getCalendarId();
-        const eventsList = await this.api.getEventsList(calendarId, '');
-        return eventsList;
-    }
-
     private organizeGuests(creator: ICreator, attendees: Array<IAttendee>) {
+        if (!attendees || attendees.length <= 0) {
+            return `${creator.email} (organizer)`;
+        }
+
         const attendeesNames = attendees
             .filter(
                 // tslint:disable-next-line: no-shadowed-variable
                 (attendees: IAttendee) =>
-                    attendees.displayName !== creator.displayName &&
-                    !attendees.self,
+                    attendees.email !== creator.email || !attendees.self,
             )
             .map(
                 // tslint:disable-next-line: no-shadowed-variable
-                (attendees: IAttendee) =>
-                    attendees.displayName ?? attendees.email,
+                (attendees: IAttendee) => attendees.email,
             );
 
-        if (attendeesNames.length <= 4) {
-            return `${creator.displayName} (organizer),${attendeesNames.toString()}`
-                .replace(
-                new RegExp(/\,/, 'g'), ', ');
+        if (attendeesNames.length > 0) {
+            if (attendeesNames.length <= 4) {
+                return `${
+                    creator.email
+                } (organizer),${attendeesNames.toString()}`.replace(
+                    new RegExp(/\,/, 'g'),
+                    ', ',
+                );
+            }
+
+            return `${creator.email} (organizer),${attendeesNames[0]},${
+                attendeesNames[1]
+            },${attendeesNames[2]} & ${attendeesNames.length - 4}
```

Comment:
Since there are three attendees names listed manually, the correct count for the rest of them would be `length - 3`, right?

```suggestion
            },${attendeesNames[2]} & ${attendeesNames.length - 3} others
```

Reply:
In this case the creator is also listed on the attendees